chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -374,7 +374,21 @@ case class ChainHandler(
 
   /** @inheritdoc */
   override def getBestFilterHeader(): Future[Option[CompactFilterHeaderDb]] = {
-    filterHeaderDAO.getBestFilterHeader
+    val bestFilterHeadersInChains = blockchains.map { blockchain =>
```

Comment:
For this to be a truely general solution, we will probably need some sort of recursive step here to look at previous headers if we zero matches in the `blockchains` we have in memory.

Reply:
Thinking about this more, i think we can do something clever with `compactFilterHeaderDAO.maxHeight` to avoid the need to endlessly iterate through the blockchain.